### DISCUSSION TITLE
Lockfile maintenance branch not automerged

### DISCUSSION THREAD

**Discussion by peterjuras** [renovatebot/renovate#12788] [Request Help]
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### Describe the bug

I have lockfile maintenance enabled together with automerge, but sometimes it does not merge the lockfile maintenance branch into the default branch, even after all checks have been run successfully. This appears to happen on multiple repositories, but only sometimes - since sometimes everything works as expected.

## Example:
(Feel free to check out the [repository](https://github.com/peterjuras/time-sync))

### Renovate config:
```json
{
  "extends": ["config:base", "group:allNonMajor"],
  "schedule": "before 7am on Friday",
  "lockFileMaintenance": {
    "enabled": true,
    "automerge": true,
    "automergeType": "branch"
  },
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch"],
      "matchCurrentVersion": "!/^0/",
      "automerge": true,
      "automergeType": "branch"
    }
  ]
}
```
### Branches
On Monday morning after renovate lockfile maintenance is performed by default:
![image](https://user-images.githubusercontent.com/3640638/142813222-bf0a93ea-1070-463f-954b-1a4b5b33fe8c.png)

### Relevant debug logs

Let me know if you need specific debug logs, all jobs on the repository completed successfully, but maybe the problem was that the subsequent job that would have merged lockfile updates was started too late. Can I somehow ensure that renovate runs more often? Or should I extend the lockfileMaintenance window?

These were the jobs on Monday (lockfile maintenance defaults to Monday before 5 AM):

* **00:07:10** - Although it should be within the lockfile maintenance schedule, the logs mention that they are skipped:
```
DEBUG: Checking schedule(before 5am on monday, null)(branch="renovate/lock-file-maintenance")
DEBUG: Checking 1 schedule(s)(branch="renovate/lock-file-maintenance")
DEBUG: Checking schedule "before 5am on monday"(branch="renovate/lock-file-maintenance")
{
  "parsedSchedule": {
    "schedules": [
      {
        "t_b": [
          18000
        ],
        "d": [
          2
        ]
      }
    ],
    "exceptions": [],
    "error": -1
  }
}
DEBUG: Package not scheduled(branch="renovate/lock-file-maintenance")
DEBUG: Skipping PR creation out of schedule(branch="renovate/lock-file-maintenance")
```
* **02:42:14** - Lockfile maintenance works and the branch is created - checks still have to be run therefore it is not automerging yet
* **06:15:35** - Lockfile maintenance is not run, because it is not within the schedule.

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by rarkins** [comment]
1. You can't ensure Renovate runs more often
2. Set `platformAutomerge=true` in your config to enable GitHub's own automerge functionality and you should see it merged as soon as the branch tests are "green"

**Reply by peterjuras** [reply]
Thanks for the quick response.

What about the first build shortly after midnight that didn't run the lockfile maintenance steps?

**Reply by rarkins** [reply]
What job ID was that and what method did you use to determine the exact time?

**Reply by peterjuras** [reply]
Job id was #512663965. The time is displayed on hover over the Date column in the dashboard.

**Reply by rarkins** [reply]
The job started at `2021-11-21 23:06:57.993443+00` so most likely explanation was that the Date check was done before the hour changed.
